### DISCUSSION TITLE
jQuery size

### DISCUSSION THREAD

**Discussion by lyusupov** [nakhonthai/ESP32APRS_T-TWR#9] [General]
I can see that you've added jQuery in between 0.2 and 0.3
This caused a sudden increase (approx. 100K) in flash memory usage.

My suggestion is:
1) transfer jQuery from webservice.cpp to a separate library
2) use gzip to compress the jQuery script into 2-3 times smaller file. Most web browsers on the market can handle compressed JavaScript files.

We use this approach in SoftRF project for many years.
Please, take a look at this file as an example:
https://github.com/lyusupov/SoftRF/blob/master/software/firmware/source/libraries/jQuery/jquery_min_js.h

Use case:
- https://github.com/lyusupov/SoftRF/blob/master/software/firmware/source/SoftRF/src/ui/Web.cpp#L56
- https://github.com/lyusupov/SoftRF/blob/master/software/firmware/source/SoftRF/src/ui/Web.cpp#L1264-L1281

**Comment by nakhonthai** [comment]
Good idea, Thank you.
